src/core/context/utils.ts

```diff
@@ -226,17 +226,19 @@ export async function doChat(
         messages: messagesWithoutReasoningContent,
         tools: tools,
         stream: true,
+        stream_options: { include_usage: modelConfig.includeUsageInStream },
       },
       {
         signal: controller.signal,
       },
     );
-
     let content = '';
     let thinkingEnd = -1;
+    let hasCloseTag = false;
+    let openTagFound = false;
```

Comment:
Qwen can start without an opening think tag, but has a closing tag that we need to look for.